### PR TITLE
Support for LockBehavior

### PR DESCRIPTION
**Pull request by thomhurst**
Support for LockBehavior

### CODE REVIEW

File: src/Sharpliner/AzureDevOps/LockBehaviour.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿using System.Diagnostics.CodeAnalysis;
+
+namespace Sharpliner.AzureDevOps;
+
+[SuppressMessage("ReSharper", "InconsistentNaming")]
+public enum LockBehaviour
+{
+    runLatest,
+    sequential
```
Comment (premun) [inline src/Sharpliner/AzureDevOps/LockBehaviour.cs:9]: In other parts of the library, we stayed more C#-like and kept naming capitalized. I am wondering whether that would not be better.

But I understand it's complicated here as this is wrapped in the `Conditioned<>` later.

Can I ask you to add a following serialization helper?

```csharp
public class YamlStringEnumConverter : IYamlTypeConverter
{
    public bool Accepts(Type type) => type.IsEnum;

    public object ReadYaml(IParser parser, Type type) => throw new NotImplementedException();

    public void WriteYaml(IEmitter emitter, object? value, Type type)
    {
        if (value is null)
        {
            emitter.Emit(new Scalar(null!));
            return;
        }

        var enumMember = type.GetMember(value.ToString()!).FirstOrDefault();
        var yamlValue = enumMember?
            .GetCustomAttributes<YamlMemberAttribute>(true)
                .Select(ema => ema.Alias)
                .FirstOrDefault()
            ?? value.ToString();
        
        emitter.Emit(new Scalar(yamlValue!));
    }
}
```

You can do following then:
```csharp
public enum LockBehaviour
{
    [YamlMember(Alias = "runLatest")]
    RunLatest,

    [YamlMember(Alias = "sequential")]
    Sequential
}
```

Comment (premun) [reply src/Sharpliner/AzureDevOps/LockBehaviour.cs:9]: I will then later apply to other places in the code that could use this.

Comment (premun) [reply src/Sharpliner/AzureDevOps/LockBehaviour.cs:9]: Oh, and register it as well:

```
    private static ISerializer InitializeSerializer()
    {
        var serializerBuilder = new SerializerBuilder()
            .WithNamingConvention(CamelCaseNamingConvention.Instance)
            .WithTypeConverter(new YamlStringEnumConverter())      <<<<<<<<<<<<<<<
            .ConfigureDefaultValuesHandling(DefaultValuesHandling.OmitDefaults | DefaultValuesHandling.OmitEmptyCollections)
            .WithEventEmitter(nextEmitter => new MultilineStringEmitter(nextEmitter));

        return serializerBuilder.Build();
    }
```

Comment (thomhurst) [reply src/Sharpliner/AzureDevOps/LockBehaviour.cs]: Nice one - Looks all good, tests seem happy

**Review by premun — APPROVED** [review]

**Pull request merged**